src/coprocessor/endpoint.rs

```diff
@@ -485,33 +486,107 @@ impl<E: Engine> Endpoint<E> {
     #[inline]
     pub fn parse_and_handle_unary_request(
         &self,
-        req: coppb::Request,
+        mut req: coppb::Request,
         peer: Option<String>,
     ) -> impl Future<Output = MemoryTraceGuard<coppb::Response>> {
         let tracker = GLOBAL_TRACKERS.insert(::tracker::Tracker::new(RequestInfo::new(
             req.get_context(),
             RequestType::Unknown,
             req.start_ts,
         )));
+        let batch_task_num = req.tasks.len();
+        let mut batch_fut_results = Vec::with_capacity(batch_task_num + 1);
+        let mut batch_responses = Vec::with_capacity(batch_task_num);
+        let mut batch_extra_info = Vec::with_capacity(batch_task_num);
+        let batch_reqs: Vec<coppb::Request> = req
+            .take_tasks()
+            .iter_mut()
+            .map(|task| {
+                let mut new_req = req.clone();
+                new_req.ranges = task.take_ranges();
+                let mut new_context = new_req.take_context();
+                new_context.set_region_id(task.get_region_id());
+                new_context.set_region_epoch(task.take_region_epoch());
+                new_context.set_peer(task.take_peer());
+                new_req.set_context(new_context);
+                batch_responses.push(coppb::StoreBatchTaskResponse::new());
+                new_req
+            })
+            .collect();
+        let mut index = 0;
+        for cur_req in batch_reqs.into_iter() {
+            let cur_tracker = GLOBAL_TRACKERS.insert(::tracker::Tracker::new(RequestInfo::new(
+                cur_req.get_context(),
+                RequestType::Unknown,
+                cur_req.start_ts,
+            )));
+            set_tls_tracker_token(cur_tracker);
+
+            match self.parse_request_and_check_memory_locks(cur_req, peer.clone(), false) {
+                Ok((handler_builder, req_ctx)) => {
+                    batch_fut_results.push(self.handle_unary_request(req_ctx, handler_builder));
+                    batch_extra_info.push((index, cur_tracker));
+                }
+                Err(e) => {
+                    make_error_batch_response(&mut batch_responses[index], e);
+                }
+            }
+            batch_responses[index].task_id = index as u64;
```

Comment:
Should we set the `task_id` in the request to the one in the response? (Then, it should be set in L512)